src/compiler/checker.ts

```diff
@@ -43172,28 +43181,15 @@ namespace ts {
                     return grammarErrorOnNode(assertionArgument, Diagnostics.Dynamic_import_only_supports_a_second_argument_when_the_module_option_is_set_to_esnext);
                 }
             }
-            if (nodeArguments.length !== 1) {
-                return grammarErrorOnNode(node, Diagnostics.Dynamic_import_must_only_have_a_specifier_and_an_optional_assertion_as_arguments);
-            }
-            return false;
-        }
-
-        function checkGrammarImportCallExpression(node: ImportCall): boolean {
-            if (moduleKind === ModuleKind.ES2015) {
-                return grammarErrorOnNode(node, Diagnostics.Dynamic_imports_are_only_supported_when_the_module_flag_is_set_to_es2020_esnext_commonjs_amd_system_or_umd);
-            }
 
-            if (node.typeArguments) {
-                return grammarErrorOnNode(node, Diagnostics.Dynamic_import_cannot_have_type_arguments);
+            if (nodeArguments.length  === 0 || nodeArguments.length > 2) {

```

Comment:
Nit: remove extra space
```suggestion
            if (nodeArguments.length === 0 || nodeArguments.length > 2) {
```